### PR TITLE
Add the possibility to edit coordinates and centre the map

### PR DESCRIPTION
**Pull request by MohamedHamouGisaia**
image

image

### CODE REVIEW

**Review by mbarbet — APPROVED** [review]

File: projects/arlas-components/assets/i18n/en.json [hunk 1]
```diff
@@ -84,5 +84,8 @@
 	"Enter a coordinate in decimal (1.1) or sexagesimal (1° 6' 3\")": "Enter a coordinate in decimal (1.1) or sexagesimal (1° 6' 3\")",
 	"Decimal: 1.1 or Sexagesimal 1°6'3\" coordinate": "Decimal: 1.1 or Sexagesimal: 1°6'3\" coordinate",
 	"Both corners have the same latitudes, change one of them.": "Both corners have the same latitudes, modify one of them.",
-	"Generate AOI": "Generate the bbox"
+	"Generate AOI": "Generate the bbox",
+	"1.1 or 1°6'3\"": "1.1 or 1°6'3\"",
+	"Move to": "Centre",
+	"Enter coordinates and centre the map": "Enter coordinates and centre the map"
```
Comment (QuCMGisaia) [inline projects/arlas-components/assets/i18n/en.json:90]: centre -> center

File: projects/arlas-components/src/lib/components/mapgl/coordinates/coordinates.component.html [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+              <mat-label>{{'Longitude'}}</mat-label>
+              <input class="input"  matInput [formControl]="coordinatesForm.longitude" [placeholder]="placeHolder | translate">
+              <mat-error *ngIf="coordinatesForm.longitude.invalid">{{getErrorMessage(coordinatesForm.longitude) | translate}}</mat-error>
+          </mat-form-field>
+      </div>
+      <button class="move" mat-stroked-button (click)="moveToCoordinates()"
+            [disabled]="coordinatesForm.invalid || coordinatesForm.latitudeErrors">
+            <span class="label">{{'Move to' | translate}}</span>
+        </button>
+  </div>
```
Comment (QuCMGisaia) [inline projects/arlas-components/src/lib/components/mapgl/coordinates/coordinates.component.html:26]: The indentation is not correct around here

File: projects/arlas-components/src/lib/components/mapgl/mapgl.component.html [hunk 3]
```diff
@@ -33,9 +33,8 @@
   </div>
 </div>
 
-<div class="current-coordinate" *ngIf="displayCurrentCoordinates">
-  <span class="lat"> Lat : {{currentLat | number:'1.5-5'}} </span>
-  <span class="lng"> Lng : {{currentLng | number:'1.5-5'}} </span>
+<div *ngIf="displayCurrentCoordinates">
```
Comment (QuCMGisaia) [inline projects/arlas-components/src/lib/components/mapgl/mapgl.component.html:36]: Could be ng-container to limit the amount of elements in the DOM

File: projects/arlas-components/src/lib/tools/coordinates.tools.ts [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+    public latitude: FormControl;
+    public longitude: FormControl;
+
+    public latitudeChanges$: Observable<any>;
+    public longitudesChanges$: Observable<any>;
+
+    public constructor(initLat: number | string, initLng: number | string) {
+        // eslint-disable-next-line max-len
+        const coordinatesRegex = '^(?<decimal>[+-]?([0-9]*[.])?[0-9]+)$|^(?<degrees>(-?)[0-9]+)°[ ]*((?<minutes>[0-9]+)\'[ ]*((?<seconds>[0-9]+)\")?)?$';
```
Comment (sebbousquet) [inline projects/arlas-components/src/lib/tools/coordinates.tools.ts:33]: Maybe the regex could be defined in the tools as a specific variable, to be used in each class without duplicate it

**Review by sebbousquet — APPROVED** [review]

File: projects/arlas-components/src/lib/tools/coordinates.tools.ts (hunk 4 above)
Comment (QuCMGisaia) [inline projects/arlas-components/src/lib/tools/coordinates.tools.ts:33]: Could be exported as a const to be defined only once

**Review by QuCMGisaia — APPROVED** [review]

File: projects/arlas-components/src/lib/tools/coordinates.tools.ts (hunk 4 above)
Comment (MohamedHamouGisaia) [reply projects/arlas-components/src/lib/tools/coordinates.tools.ts]: fixed

Comment (MohamedHamouGisaia) [reply projects/arlas-components/src/lib/tools/coordinates.tools.ts]: fixed

File: projects/arlas-components/src/lib/components/mapgl/mapgl.component.html (hunk 3 above)
Comment (MohamedHamouGisaia) [reply projects/arlas-components/src/lib/components/mapgl/mapgl.component.html]: fixed

File: projects/arlas-components/assets/i18n/en.json (hunk 1 above)
Comment (MohamedHamouGisaia) [reply projects/arlas-components/assets/i18n/en.json:90]: british spelling

**Pull request merged**